torchtune/modules/loss/ppo.py

```diff
@@ -0,0 +1,98 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Optional, Tuple
+
+import torch
+import torch.nn as nn
+from torchtune.utils import ppo_utils
+
+
+class PPOLoss(nn.Module):
```

Comment:
If this loss does not have any trainable parameters I would be more inclined to keep it as a function (it's possible that I'm missing something obvious here though, if so please correct me)

Reply:
there's no trainable parameters - this was mainly to ensure consistency with the other configs in that there's a callable loss component which is instantiated. in practice there's no other loss you're going to be using with this recipe though, so we can probably drop this format here

Reply:
Ah right, the instantiation logic.. in that case I am OK to leave it like this. Actually for this kind of stuff what I like to do is keep the loss as a function and do something like this in the config

```
loss:
  _component_: functools.partial
  func: torchtune.modules.loss.PPOLoss
  gamma: 1
  lmbda: 0.95
  epsilon: 0.2
  value_coeff: 0.1
  value_clip_range: 0.1
```

Caveat: I haven't actually tried this in the torchtune config system so it may not actually work. Also more importantly I think most people will find this unintuitive.

Reply:
I personally like this suggestion, but think it's easier to interpret to keep it as a class. If we don't like inheriting from `nn.Module`, I could just replace `forward` with `__call__`? Not sure if that would also be confusing to some people.